### PR TITLE
win_owner: added tests and fixed up bool handling

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
Added integration tests for win_owner, fixed up docs with recurse parameter and also fixed up the handling of this parameter so it can take in`true/false` as well as `yes/no`.

##### ISSUE TYPE
 - Feature Pull Request
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME
win_owner

##### ANSIBLE VERSION
```
ansible 2.4.0 (devel 6a3ce18e4b) last updated 2017/05/25 20:15:42 (GMT +1000)
  config file =
  configured module search path = [u'/home/jordan/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /mnt/c/dev/linux-ansible/lib/ansible
  executable location = /mnt/c/dev/linux-ansible/bin/ansible
  python version = 2.7.13 (default, Apr 13 2017, 08:16:29) [GCC 5.4.0 20160609]
```

### CODE REVIEW

**Comment by gundalow** [comment]
I wonder if `Get-AnsibleParam`  should throw an error if `validateset` and type bool are specified

File: test/integration/test_win_group1.yml [hunk 1]
```diff
@@ -9,3 +9,4 @@
     - { role: win_fetch, tags: test_win_fetch }
     - { role: win_regmerge, tags: test_win_regmerge }
     - { role: win_regedit, tags: test_win_regedit }
+    - { role: win_owner, tags: test_win_owner }
```
Comment (dagwieers) [inline test/integration/test_win_group1.yml]: AFAIK you don't have to do this yourself.

**Review by dagwieers — APPROVED** [review]
LGTM shipit

File: test/integration/test_win_group1.yml (hunk 1 above)
Comment (jborean93) [reply test/integration/test_win_group1.yml]: Would be good to know what to do for this

@mattclay are you able to clarify?

**Comment by jborean93** [comment]
@dagwieers I've removed the manual entry for test_win_group_1.yml

**Comment by jborean93** [comment]
bot_status

**Comment by dagwieers** [comment]
@jborean93 My shipits don't count, so I am no longer doing those.

File: test/integration/test_win_group1.yml (hunk 1 above)
Comment (mattclay) [reply test/integration/test_win_group1.yml]: The `test_win_group` prefixed files are used for the legacy method of running Windows tests and don't need to be updated. They probably won't be around much longer.

**Comment by jborean93** [comment]
Thanks @jhawkesworth for the merge.

**Pull request merged**